tests/kafkatest/tests/core/snapshot_test.py

```diff
@@ -0,0 +1,227 @@
+        wait_until(lambda: self.file_exists(node, file_path) == True,
+                   timeout_sec=100,
+                   backoff_sec=1, err_msg="Not able to verify snapshot existence in a reasonable amount of time")
+
+    def choose_controller_node(self):
+        if self.kafka.remote_controller_quorum:
+            # We are running in remote kraft
+            controller_node = random.choice(self.kafka.remote_controller_quorum.nodes)
+        else:
+            # This is colocated kraft
+            # Picking the first node as that is guaranteed to be a controller
+            controller_node = self.kafka.nodes[0]
+
+        self.logger.debug("Picking the controller node %s" % self.kafka.who_am_i(controller_node))
+        return controller_node
+
+    def file_exists(self, node, file_path):
+        # Check if the first log segment is cleaned up
+        self.logger.debug("Checking if file %s exists"%file_path)
+        cmd = "ls %s" % file_path
+        files = node.account.ssh_output(cmd, allow_fail=True, combine_stderr=False)
+
+        if len(files) is 0:
+            self.logger.debug("File %s does not exist"%file_path)
+            return False
+        else:
+            self.logger.debug("File %s was found"%file_path)
+            return True
+
+    def validate_success(self):
+        cluster_id = self.kafka.cluster_id()
+        assert cluster_id is not None
+        assert len(cluster_id) == 22
+        assert self.kafka.all_nodes_support_topic_ids()
+        assert self.kafka.check_protocol_errors(self)
+
+
+    @cluster(num_nodes=9)
+    @parametrize(metadata_quorum=quorum.remote_kraft)
+    # TODO: enable after concurrent startup has been merged
+    #@matrix(metadata_quorum=quorum.all_kraft)
+    def test_broker(self, metadata_quorum=quorum.colocated_kraft):
+        # Scenario -- stop a broker node and create metadata changes.
+        # Restart the broker and let it catch up
+        node = random.choice(self.kafka.nodes)
+        self.logger.debug("Scenario 1: kill-create_topics-start on node %s", self.kafka.who_am_i(node))
+        self.kafka.stop_node(node)
+        # Now modify the cluster to create more metadata changes
+        self.topics_created += self.create_n_topics(topic_count=10)
+        self.kafka.start_node(node)
+        self.validate_success()
+
+        # Scenario -- Re-init broker after cleaning up all persistent state
+        node = random.choice(self.kafka.nodes)
+        self.logger.debug("Scenario 2: kill-clean-start on node %s", self.kafka.who_am_i(node))
+        self.kafka.clean_node(node)
+        self.kafka.start_node(node)
+        self.validate_success()
+
+        # Scenario -- Re-init broker after cleaning up all persistent state
+        # and create a topic where the affected broker must be the leader
+        node = random.choice(self.kafka.nodes)
+        self.logger.debug("Scenario 3: kill-clean-start-verify-produce on node %s", self.kafka.who_am_i(node))
```

Comment:
Do we think that we need these 3 scenarios in this test? Do you think that we only need this scenario to verify that snapshot are getting generated and loaded by the brokers?